test/functional/basic/find-e2e-specs.js

```diff
@@ -400,4 +404,53 @@ describe('XCUITestDriver - find', function () {
       await els[0].getAttribute("type").should.eventually.eql("XCUIElementTypeTable");
     });
   });
+
+  describe('by image', function () {
+    before(async function () {
+      // use the driver settings that cause the most code paths to be exercised
+      await driver.updateSettings({
+        fixImageTemplateSize: true,
+        autoUpdateImageElementPosition: true,
+      });
+    });
+    it('should find image elements', async function () {
+      let els = await driver.elementsByImageFile(UICATALOG_LIST_IMG);
+      els.should.have.length(1);
+    });
+    it('should find an image element', async function () {
+      let el = await driver.elementByImageFile(UICATALOG_LIST_IMG);
+      el.value.should.match(/appium-image-element/);
+    });
+    it('should not find an image element that is not matched', async function () {
+      await driver.elementByImageFile(TESTAPP_BUTTON_IMG)
+        .should.eventually.be.rejectedWith(/Error response status: 7/);
+    });
+    it('should find anything with a threshold low enough', async function () {
+      const {imageMatchThreshold} = await driver.settings();
+      await driver.updateSettings({imageMatchThreshold: 0});
+      await driver.elementByImageFile(TESTAPP_BUTTON_IMG).should.eventually.exist;
```

Comment:
If, for any reason, this fails, the image match threshold will remain 0. Perhaps better to put this in a try/finally block, if not a separate describe block with a before/after.

Reply:
mmmm, good point